File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.*;

class Solution {
    public long solution(long n) {
        String str = "";
        
        int[] arr = new int[10];
        
        while(n > 0){
            int num = (int)(n % 10);
            arr[num]++;
            n /= 10;
        }
        
        for(int i=9; i >= 0; i--){
            for(int j=0; j<arr[i]; j++){
                str += i;
            }
        }
        
        return Long.parseLong(str);
    }
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(long long n) {
    vector<int> v;
    while(n != 0) {
        v.push_back(n % 10);
        n /= 10;
    }
    
    sort(v.begin(), v.end(), greater<int>());
    string str = "";
    
    for(int num : v) {
        str += to_string(num);
    }
    
    long long answer = stoll(str);
    return answer;
}
